### PR TITLE
Chore: Fix issues related to Node 12 becoming LTS

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Changes introduced in the `@wordpress/scripts` package required to make it work with Node 12:

> ### Breaking Changes
> 
> - The bundled `npm-package-json-lint` dependency has been updated from requiring `^3.6.0` to requiring `^4.0.2` ([#18054](https://github.com/WordPress/gutenberg/pull/18054)). Please see the [migration guide](https://npmpackagejsonlint.org/docs/en/v3-to-v4). Note: `npmPackageJsonLintConfig` prop in the `package.json` file needs to be renamed to `npmpackagejsonlint`.
> 
> ### New Features
> 
> - The bundled `puppeteer` dependency has been updated from requiring `^1.19.0` to requiring `^1.20.0` ([#18054](https://github.com/WordPress/gutenberg/pull/18054)). It uses Chromium v78 instead of Chromium v77.

Based on the feedback from @talldan, the root `package.json` file is now included in the linting process. The related config for linting was moved out of the `package.json` file to its own file.

This PR also enables the new rule introduced in `npm-package-json-lint` v4.x: `prefer-no-devDependencies` – to ensure there are no dev dependencies added to packages. Major props to @simison as this is going to remove some maintenance burden from reviewers.

## How has this been tested?
`npm run lintpkg-json`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -164,6 +166,7 @@
 					"requireEndingPeriod": true
 				}
 			],
+			"prefer-no-devDependencies": "error",
```
Comment (gziolo) [inline package.json]: There is new rule added which we will benefit from in Gutenberg. See https://github.com/tclindner/npm-package-json-lint/pull/133. Props to @simison for working on it 💯

File: package.json [hunk 2]
```diff
@@ -154,7 +156,7 @@
 		"webpack": "4.41.0",
 		"worker-farm": "1.7.0"
 	},
-	"npmPackageJsonLintConfig": {
+	"npmpackagejsonlint": {
```
Comment (gziolo) [inline package.json]: This one was an undocumented breaking change but explained in https://github.com/tclindner/npm-package-json-lint/issues/140. @ntwb, I see you had the same issue :)

File: package.json [hunk 3]
```diff
@@ -204,7 +207,7 @@
 		"lint-js": "wp-scripts lint-js",
 		"lint-js:fix": "npm run lint-js -- --fix",
 		"lint-php": "wp-scripts env lint-php",
-		"lint-pkg-json": "wp-scripts lint-pkg-json ./packages",
+		"lint-pkg-json": "wp-scripts lint-pkg-json 'packages/*/package.json'",
```
Comment (gziolo) [inline package.json]: This also looks like a breaking change. @tclindner - was it intended?

**Comment by gziolo** [comment]
I tried updating several packages reported by `npm audit` which depend on `https-proxy-agent`, but they still aren't updated. We should revisit it later. I can remove all the related changes triggered with `npm audit fix` efforts.

File: package.json (hunk 2 above)
Comment (ntwb) [reply package.json]: Yes, probably should have been added to the latest release notes...

**Comment by talldan** [comment]
Tested in Node 12, and it worked for me.

I didn't realise that `npm-package-json-lint` doesn't report linting errors in the main package.json. Is that expected?

**Comment by gziolo** [comment]
> I didn't realise that `npm-package-json-lint` doesn't report linting errors in the main package.json. Is that expected?

I can tweak it to include the one in the root folder. There are only 3 violations reported.

**Comment by gziolo** [comment]
> I didn't realise that `npm-package-json-lint` doesn't report linting errors in the main package.json. Is that expected?

@talldan, see d97282d. It's now linted as well.

**Review by ntwb — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (tclindner) [reply package.json]: Sorry for missing this in the docs. The change log and migration guide have been updated!

File: package.json (hunk 3 above)
Comment (tclindner) [reply package.json]: Hi @gziolo, no it was not. I've released v4.0.3 that should correct the issue. Please let me know if you have other issues!
https://github.com/tclindner/npm-package-json-lint/releases/tag/v4.0.3

Comment (gziolo) [reply package.json]: Nice, thanks. We will update the dependency right away 👍

Comment (gziolo) [reply package.json]: > Nice, thanks. We will update the dependency right away 👍

We are using ranges so we probably should be fine, @ntwb @nerrad or @swissspidy can you confirm?

Comment (ntwb) [reply package.json]: The _root_ lock file needs updating for this, currently:
```json
24344          "npm-package-json-lint": {
24345                  "version": "4.0.2",
...
```

Comment (gziolo) [reply package.json]: I opened https://github.com/WordPress/gutenberg/pull/18160 to address it.

**Pull request merged**